### DISCUSSION TITLE
Moon Rover - Facing process priority issues running cFS in Linux

### DISCUSSION THREAD

**Discussion by srini-cmu** [nasa/cFS#479] [Q&A] [answered]
Is it possible to run cFS in userspace on Linux (Ubuntu 18.04). Running cFS with root privileges causes it to trigger a bug in the camera service daemon due to the daemon running at a lower priority. 

Even if cFS runs in the kernel space, it will be useful if we can run it with lower priority so that the drivers can run at realtime (RT) priority. 

Or is it possible to assign cores to cFS so that the other cores are free for other apps/drivers? I tried `chrt` command, but wasn't very useful.

Please comment if this description is too vague. I am happy to add more details.

**Comment by skliper** [comment, marked as answer]
If you set `OSAL_CONFIG_DEBUG_PERMISSIVE_MODE` true it'll allow running without root privileges... whatever it doesn't have permissions for it'll just do best effort and continue.  See example for assigning task cpu affinity from a PSP here: https://github.com/nasa/PSP/pull/341

The cFS apps and core should all have configurable priority levels also... see the startup script and config files (anything w/ `PRIORITY`).

**Reply by srini-cmu** [reply]
Great. Thanks for those pointers. I will look into those.
